src/Library/types.ts

```diff
@@ -0,0 +1,85 @@
+export type Author = {
+    readonly name: string;
+    readonly url: string;
+    readonly position: string;
+};
+
+enum DefaultCategories {
+    All = 'All',
+    Popular = 'Popular'
+}
+
+enum PartnersCategories {
+    Reastream = 'Reastream'
+}
+
+export type LibraryCategories = DefaultCategories | PartnersCategories;
+
+export enum LibraryDirections {
+    All = 'All',
+    Graphics = 'Graphics',
+    Dev = 'Dev',
+    Marketing = 'Marketing',
+    Management = 'Management',
+    Advertising = 'Advertising'
+}
+
+export enum LibraryFormats {
+    All = 'All',
+    Conference = 'Conference',
+    Video = 'Video',
+    Tutorial = 'Tutorial',
+    Interview = 'Interview'
+}
+
+export enum LibraryLevels {
+    All = 'All',
+    Beginner = 'Beginner',
+    Specialist = 'Specialist'
+}
+
+enum VideoSize {
+    Default = 'Default',
+    Featured = 'Featured'
+}
+
+export interface Video {
+    readonly id: number;
+    readonly title: string;
+    readonly description: string;
+    readonly mainCategory?: LibraryCategories;
```

Comment:
It's a very minor comment
`mainCategory` is a singular, but `LibraryCategories` is a plural.
It might be better to use singulars for `enums`, for example `LibraryCategory`
If you're expecting a list you are writing `LibraryCategory[]` and it's clear.

Reply:
Good point, thanks.